src/library_syscall.js

```diff
@@ -446,6 +446,7 @@ var SyscallsLibrary = {
   },
   __syscall91: function(which, varargs) { // munmap
     var addr = SYSCALLS.get(), len = SYSCALLS.get();
+    if (addr < 0 || len == 0) return -ERRNO_CODES.EINVAL;
```

Comment:
why would `addr` be less than zero? (is this a check for an address greater than 2GB?)

Reply:
This check was based on munmap/8-1 test from POSIX test suite which passes -1 as `addr` argument and expects EINVAL.

I'll reconsider this check.

Reply:
Changed to check addr != MAP_FAILED in https://github.com/emscripten-core/emscripten/pull/8633/commits/43fcce7707af84dfe995df60a4fd80e8456e6fad